pkg/stanza/fileconsumer/config.go

```diff
@@ -109,6 +121,10 @@ func (c Config) buildManager(logger *zap.SugaredLogger, emit EmitFunc, factory s
 	default:
 		return nil, fmt.Errorf("invalid start_at location '%s'", c.StartAt)
 	}
+	var deleteAfterRead bool
+	if !featuregate.GetRegistry().IsEnabled(allowFileDeletion) {
+		deleteAfterRead = c.DeleteAfterRead
+	}
```

Comment:
I think this condition is inverted (deleteAfterRead should only be set if the featuregate is enabled)